docs/my_documentation.rst

```diff
@@ -0,0 +1,544 @@
+		'oauth': _github_oauth_config,
+		'endpoint': '/user',
+		'identifier_field': 'id',
+		'mapping': {
+		    'user_name': 'login',
+		    'affiliation': 'company'
+		}
+	    }
+	}
+	
+Important thing to notice here is assigning the ``'identifier_field'`` to name of the field containing unique user identifier of the external provider. In this GitHub example its ``'id'``.  This ``'identifier_field'`` is used to connect the identity to the user, so make sure you assign it to the right unique field.
+
+Next thing to notice is mapping of fields from our application to those of the provider. In this example we map ``'user_name'`` - that we use in our application to ``'login'`` field used in GitHub.
+
+Also make sure to connect the authentication provider to identity provider in provider mapping.
+
+.. code-block:: python
+
+	MULTIPASS_PROVIDER_MAP = {
+	    'github': 'github'
+	}
+	
+In this example we only have one provider but if you have more than one provider you need to pass a template file to ``MULTIPASS_LOGIN_SELECTOR_TEMPLATE``. In this template users should choose which provider they want to log in with.
+
+.. code-block:: python
+
+	MULTIPASS_LOGIN_SELECTOR_TEMPLATE = 'login_selector.html'
+	
+	
+This configuration can be added to your flask configuration file that you use when initializing flask application. (as shown in < Initialization>) However, you can configure multipass also directly through application object. for example:
+
+.. code-block:: python
+
+	app.config['MULTIPASS_LOGIN_SELECTOR_TEMPLATE'] =  'login_selector.html'
+
+Login
+---------------------
+The easiest way to process login is to specify your prefered login URLs in configuration as value for 	``MULTIPASS_LOGIN_URLS``. For example:   
+
+.. code-block:: python
+
+	MULTIPASS_LOGIN_URLS = {'/my_login/', '/my_login/<provider>'}
+	
+(By default,  values ``'/login/'`` and ``'/login/<provider>'`` are set for ``MULTIPASS_LOGIN_URLS`` so if those suits your application there's no need to change them.)
+	
+Multipass then binds  ``process_login`` method to these URLs upon initialization of multipass.
+
+However,  if you wish to execute some additional code before, make ``MULTIPASS_LOGIN_URLS`` empty,  handle login request yourself and then call ``process_login``on your  ``Multipass`` object:
+
+.. code-block:: python
+
+	@app.route('/my_login/', methods=('GET', 'POST'))
+	@app.route('/my_login/<provider>', methods=('GET', 'POST'))
+	def login(provider=None):
+	    # Your additional code
+	    return multipass.process_login(provider)
+	
+
+When calling ``process_login`` with provider name as an argument the ``initiate_external_login`` on the provider gets called and redirects user to the provider's site to perform login.( In case of local provider <``_login_form`` link to login of local providers> gets called.) If provider is not specified the ``process_login`` redirects to a login selection template specified in configuration as ``MULTIPASS_LOGIN_SELECTOR_TEMPLATE``. 
```

Comment:
And check the Sphinx documentation on [`:ref:`](http://www.sphinx-doc.org/en/stable/markup/inline.html#role-ref) in order to create a proper link.